pandas/core/internals.py

```diff
@@ -2684,6 +2705,19 @@ def shift(self, periods, axis=0, mgr=None):
         return [self.make_block_same_class(new_values,
                                            placement=self.mgr_locs)]
 
+    def concat_same_type(self, to_concat):
```

Comment:
we should prob make a dispatch function in ``_concat`` to avoid all of this boilerplate

``def _concat_for_type(to_concat, typ='sparse|datetime|cat')``